gems/pending/util/extensions/miq-file.rb

```diff
@@ -52,7 +53,7 @@ def self.sizeEx(path)
 
   def self.path_to_uri(file, hostname = nil)
     hostname ||= MiqSockUtil.getFullyQualifiedDomainName
-    URI.join("file://#{hostname}", "/#{URI.encode(file.tr('\\', '/'))}").to_s
+    URI.join("file://#{hostname}", "/#{Addressable::URI.encode(file.tr('\\', '/'))}").to_s
```

Comment:
@djberg96  Do we care about IPv6 addresses for hostnames?  I'd assume file URIs would also need to have them in square brackets?

```ruby
irb(main):002:0> hostname = "127.0.0.1"
irb(main):015:0> URI.join("file://myhost", "/#{Addressable::URI.encode(file.tr('\\', '/'))}").to_s
=> "file://myhost/foo/C:/bar"
irb(main):016:0> URI.join("file://::1", "/#{Addressable::URI.encode(file.tr('\\', '/'))}").to_s
URI::InvalidURIError: bad URI(is not URI?): file://::1
```

vs.

```ruby
irb(main):018:0> URI::Generic.build(:scheme => "file", :host => "myhost", :path => "/#{Addressable::URI.encode(file.tr('\\', '/'))}").to_s
=> "file://myhost/foo/C:/bar"
irb(main):019:0> URI::Generic.build(:scheme => "file", :host => "::1", :path => "/#{Addressable::URI.encode(file.tr('\\', '/'))}").to_s
=> "file://[::1]/foo/C:/bar"
```